jsapp/js/components/submissions/submissionUtils.tsx

```diff
@@ -777,3 +778,53 @@ export function markAttachmentAsDeleted(
 
   return data
 }
+
+/**
+ * Removes empty objects (and arrays) from the given object recursively.
+ * This function mutates the original object.
+ */
+export function removeEmptyObjects(obj: { [key: string]: any }) {
+  if (typeof obj !== 'object' || obj === null) {
+    return obj
+  }
+  // Recursively process each property
+  for (const key in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, key)) {
+      obj[key] = removeEmptyObjects(obj[key])
+      // Remove the property if it is an empty object
+      if (typeof obj[key] === 'object' && obj[key] !== null && Object.keys(obj[key]).length === 0) {
+        // This is a safer way to do `delete obj[key]`:
+        obj = Object.fromEntries(Object.entries(obj).filter(([objKey]) => objKey !== key))
+      }
+    }
+  }
+  return obj
+}
+
+/**
+ * This function removes all possible empty objects from given submission supplemental details. If there were only empty
+ * objects in it (nested), you can end up with an empty object as an final outcome.
+ */
+export function removeEmptyFromSupplementalDetails(supplementalDetails: SubmissionSupplementalDetails) {
+  let details = clonedeep(supplementalDetails)
+
+  // Step 1: Remove responses to qual questions that are:
+  // a) empty strings (means "no response" or "response deleted")
```

Comment:
Okay looks like the function is supposed to solve my earlier comment about case 7. This is how I reproduced it:
1. Create a qual number question
2. Answer the question
3. Backspace the answer
4. See that Open button still is appearing in the table

Let me know if I misunderstood how this case works or if this is a bug :)

Reply:
Good catch. I improved this code to also handle other "empty" qualitative analysis question responses.